### PR TITLE
V3: Assign deps to assets

### PR DESCRIPTION
**Pull request by alshdavid**
This PR adds dependencies to assets by relating them during the Rust AssetGraph conversion

### CODE REVIEW

File: packages/core/core/src/requests/AssetGraphRequestRust.js
Comment (alshdavid) [inline packages/core/core/src/requests/AssetGraphRequestRust.js:1]: If the `Asset` is seen first then create a `Map` to contain the `Dependencies` and assign a reference to it to the `Asset.dependencies`.

If a `Dependency` is seen first, create a `Map` to contain `Dependencies` and assign a reference to the `Dependency` there. The `Asset` will collect the reference later.

File: packages/core/core/src/requests/AssetGraphRequestRust.js [hunk 1]
```diff
@@ -234,6 +250,8 @@ function getAssetGraph(serializedGraph, options) {
     }
   }
 
+  dependencyMap.clear();
```
Comment (mattcompiles) [inline packages/core/core/src/requests/AssetGraphRequestRust.js:253]: Why the clear?

Comment (alshdavid) [reply packages/core/core/src/requests/AssetGraphRequestRust.js:253]: Not really needed and I can remove it if it's ambiguous/not actually valuable.

My thinking was that we should probably save memory where we can and that map will be filled with the string values of every asset from the Asset Graph - clearing it to give the GC a head start.

Though the dependencyMap falls out of scope a second later so it's probably not valuable 😆

Comment (alshdavid) [reply packages/core/core/src/requests/AssetGraphRequestRust.js]: Removed

**Review by mattcompiles — APPROVED** [review]

**Pull request merged**